crates/core/src/delta_datafusion/expr.rs

```diff
@@ -276,33 +272,6 @@ impl<'a> Display for SqlFormat<'a> {
                     write!(f, "{expr} IN ({})", expr_vec_fmt!(list))
                 }
             }
-            Expr::GetIndexedField(GetIndexedField { expr, field }) => match field {
```

Comment:
Why is this being removed?

Reply:
Those functions don't exist before, and based on my understanding from https://github.com/apache/datafusion/pull/10568, those rewrites happen earlier now so the functionality stays the same, it's just done by DF

Reply:
Just wondering if we can catch the new functions and keep the old display formatting

Reply:
Hmmmm, I'm not entirely sure this is possible. Those operations have been converted into ScalarFunctions, so the concept of "taking an index of a field" is now "a scalar function that operates on a field and an index". Had to modify a bunch of tests to accommodate. https://github.com/delta-io/delta-rs/blob/af5997cf7dd8cce97b07b5c8c78e23cb7092549f/crates/core/src/delta_datafusion/expr.rs#L656-L659

It could be possible manually match on scalar functions and convert them into the old style, though.

Reply:
Hmm, it would make the roundtrippable format closer to SQL, but I guess with the timestamp parsing I already kind of stopped doing that 😆